services/httpd/handler_test.go

```diff
@@ -318,6 +318,49 @@ func TestHandler_Ping(t *testing.T) {
 	}
 }
 
+// Ensure the handler returns the version correctly from the different endpoints.
+func TestHandler_Version(t *testing.T) {
+	h := NewHandler(false)
+	w := httptest.NewRecorder()
+	tests := []struct {
+		method   string
+		endpoint string
+		body     *bytes.Reader
+	}{
+		{
+			method:   "GET",
+			endpoint: "/ping",
+			body:     nil,
+		},
+		{
+			method:   "GET",
+			endpoint: "/query?db=foo&q=SELECT+*+FROM+bar",
+			body:     nil,
+		},
+		{
+			method:   "POST",
+			endpoint: "/write",
+			body:     bytes.NewReader(make([]byte, 10)),
+		},
+	}
+
+	for _, test := range tests {
+		if test.body != nil {
```

Comment:
@rossmcdonald  you should just be able to do 

```go
h.ServeHTTP(w, MustNewRequest(test.method, test.endpoint, test.body)
```
regardless of whether `test.body` is `nil` or not.

Maybe ping me over the stacktrace when you do use the single line and I can take a look it you like?